assets/js/blocks/attribute-filter/block.js

```diff
@@ -0,0 +1,181 @@
+/**
+ * External dependencies
+ */
+import {
+	useCollection,
+	useQueryStateByKey,
+	useQueryStateByContext,
+} from '@woocommerce/base-hooks';
+import {
+	useCallback,
+	Fragment,
+	useEffect,
+	useState,
+	useMemo,
+} from '@wordpress/element';
+import { sortBy } from 'lodash';
+import CheckboxList from '@woocommerce/base-components/checkbox-list';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import { getTaxonomyFromAttributeId } from '../../utils/attributes';
+
+/**
+ * Component displaying an attribute filter.
+ */
+const AttributeFilterBlock = ( { attributes } ) => {
+	const [ options, setOptions ] = useState( [] );
+	const [ checkedOptions, setCheckedOptions ] = useState( [] );
+	const { showCounts, attributeId, queryType } = attributes;
+	const taxonomy = getTaxonomyFromAttributeId( attributeId );
+
+	const [ queryState ] = useQueryStateByContext( 'product-grid' );
+	const [ productAttributes, setProductAttributes ] = useQueryStateByKey(
+		'product-grid',
+		'attributes',
+		[]
+	);
+
+	const filteredCountsQueryState = useMemo( () => {
+		// If doing an "AND" query, we need to remove current taxonomy query so counts are not affected.
```

Comment:
Without a taxonomy this query will be broken... I think the change to the useCollection hook would be good so we can avoid the query altogether if missing (since we cannot add a conditional before the hooks). Follow up?

Reply:
+ to follow up.